org/mozilla/jss/nss/SSLFDProxy.java

```diff
@@ -19,9 +19,12 @@
     public int outboundOffset;
 
     public boolean needCertValidation;
+    public boolean needBadCertValidation;
+    public int badCertError;
     public boolean handshakeComplete;
 
-    public CertAuthHandler handler;
+    public CertAuthHandler certAuthHandler;
+    public BadCertHandler badCertHandler;
```

Comment:
Just a note, the old public `handler` field is no longer available, so this is not a backward compatible change, but it probably doesn't matter assuming it's only used internally.

Reply:
Yeah, `nss/SSLFDProxy` is used mostly internally. It has to be public as earlier discussed.